tests/test_utils/mock_executor.py

```diff
@@ -36,10 +36,17 @@ def __init__(self, do_update=True, *args, **kwargs):
         self.history = []
         # All the tasks, in a stable sort order
         self.sorted_tasks = []
-        self.mock_task_results = defaultdict(lambda: State.SUCCESS)
+
+        # If multiprocessing runs in spawn mode,
+        # arguments are to be pickled but lambda is not picclable.
+        # So we should pass self.success instead of lambda.
```

Comment:
Remove this comment please.

Reply:
Someone might want to try to replace [this](https://github.com/apache/airflow/pull/7128/files#diff-d7e6d11bfaeca087149ad577dd7704daR43) with lambda.
Do you have any other idea to avoid that without this note?

Reply:
Let me explain further. Actually, `MockExecutor` is also passed to a sub-process using multiprocessing.
So without this change, I got the same problem.

Reply:
Ohhh gotcha.

Reply:
Oddly enough I wonder what part of mock_executor is passed to a subprocess? This should all happen "in" the scheduler...

Anyway I understand this now, thanks.

Reply:
If we run `TestSchedulerJob.test_retry_still_in_executor` on macOS, we will get following stack-trace.

```
  File "/Users/kou/work/oss/airflow/airflow/jobs/scheduler_job.py", line 1499, in _execute
    self._execute_helper()
  File "/Users/kou/work/oss/airflow/airflow/jobs/scheduler_job.py", line 1532, in _execute_helper
    self.processor_agent.start()
  File "/Users/kou/work/oss/airflow/airflow/utils/dag_processing.py", line 353, in start
    self._process.start()
  File "/opt/python3/3.8.1/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/opt/python3/3.8.1/lib/python3.8/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/opt/python3/3.8.1/lib/python3.8/multiprocessing/context.py", line 283, in _Popen
    return Popen(process_obj)
  File "/opt/python3/3.8.1/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/opt/python3/3.8.1/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/opt/python3/3.8.1/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/opt/python3/3.8.1/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'MockExecutor.__init__.<locals>.<lambda>'
```

Reply:
I was wondering if we could add an automated test to make sure it all stays picklable. 

We could have a test that executes all the multiprocessing-relevant code from those Airflow classes and runs them in "spawn" mode also on Linux. https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods

I think the easiest way we could do is to simply add a dedicated environment variable  in the airflow startup code (and call `mp.set_start_method('spawn')` if the variable is set)  and process one of the example Dags with that environment variable set. 

That would be a nice protection for the future.

Reply:
I'm considering to add `MP_START_METHOD` environment variable and `mp_start_method` property in `core` section.
Should I do that in this PR?

Reply:
I think it would be best to add it here. It is very much related.

Reply:
OK, I'll do it.

Reply:
For the CI , we might have so much work to do to pass tests on travis.
So how about adding an environment variable and a property I suggested above and open another PR to work on CI stuff?

Reply:
I think I'd prefer to have it in one commit (@ashb - what do you think) ?

Reply:
OK. I've added new CI entries in .travis.yml.
Waiting for all tests pass...